Mathlib/Combinatorics/Quiver/ConnectedComponent.lean

```diff
@@ -57,11 +65,208 @@ end WeaklyConnectedComponent
 
 variable {V}
 
--- Without the explicit universe level in `Quiver.{v+1}` Lean comes up with
--- `Quiver.{max u_2 u_3 + 1}`. This causes problems elsewhere, so we write `Quiver.{v+1}`.
 /-- A wide subquiver `H` of `Symmetrify V` determines a wide subquiver of `V`, containing an
 arrow `e` if either `e` or its reversal is in `H`. -/
 def wideSubquiverSymmetrify (H : WideSubquiver (Symmetrify V)) : WideSubquiver V :=
   fun _ _ ↦ { e | H _ _ (Sum.inl e) ∨ H _ _ (Sum.inr e) }
 
+/-!
+## Strongly connected components (directed connectivity)
+We reuse `IsPreconnected V` for directed strong connectivity and add a positive-length variant.
+We then define the equivalence relation for strongly connected components and the quotient type.
```

Comment:
Comments get out of date very quickly so I'd only add them if they're really adding a lot
```suggestion
```